source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -38,10 +42,30 @@ function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
 	}
 }
 
+function markAsViewedSelector(target: HTMLElement): string {
+	const checked = (target as HTMLInputElement).checked ? ':not([checked])' : '[checked]';
+	return '.js-reviewed-checkbox' + checked;
+}
+
+const markAsViewed = clickAll(markAsViewedSelector);
+
+async function onAltClick(event: delegate.Event<MouseEvent, HTMLElement>): Promise<void> {
+	if (event.altKey && event.isTrusted) {
+		const lastCheckboxCompleted = oneMutation(select.last(markAsViewedSelector(event.target as HTMLElement))!, {attributes: true});
+		const toast = new Toast();
+		toast.show();
+		await delay(30); // Without this, the Toast doesn't appear in time
+		markAsViewed(event);
+		await lastCheckboxCompleted; // Without this, done will be called too early
```

Comment:
How early? I just checked #4044 on Chrome and I don't see a difference?

Reply:
Without it the 'completed' toast is never shown. (I just tried can provide git if needed)

Reply:
At first sight it works and I do see the toast, but also I noticed that the 3s timer starts **before** the completed toast appears because the page is still doing _layout_ or something 🤷‍♂️ so if that operation takes longer than the 3s, the completed toast is removed before it's even painted. 

This should be taken care by the toast component itself, so I moved this logic into its file.